File: m4dModels/DanceLink.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace m4dModels
{
    public class DanceLink
    {
        public Guid Id { get; set; }
        public string DanceId { get; set; }

        [DataType(DataType.MultilineText)]
        public string Description { get; set; }
        public string Link { get; set; }
    }
}

File: m4d/ClientApp/src/models/DanceLink.ts
import { jsonMember, jsonObject } from "typedjson";

@jsonObject
export class DanceLink {
  @jsonMember(String) public id!: string;
  @jsonMember(String) public danceId!: string;
  @jsonMember(String) public description!: string;
  @jsonMember(String) public link!: string;

  public constructor(init?: Partial<DanceLink>) {
    Object.assign(this, init);
  }

  public cloneAndModify(changes: Partial<DanceLink>): DanceLink {
    const clone = new DanceLink(this);
    Object.assign(clone, changes);
    return clone;
  }
}
